chromium_src/chrome/browser/ui/browser_tabrestore.cc

```diff
@@ -0,0 +1,73 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_webtorrent/browser/buildflags/buildflags.h"
+#include "chrome/browser/ui/browser_tabrestore.h"
+
+#if BUILDFLAG(ENABLE_BRAVE_WEBTORRENT)
+#include "brave/browser/extensions/brave_component_loader.h"
+#include "brave/components/brave_webtorrent/browser/webtorrent_util.h"
+#include "chrome/browser/extensions/extension_service.h"
+#include "extensions/browser/extension_system.h"
+#endif
+
+#define AddRestoredTab AddRestoredTab_ChromiumImpl
+#include "../../../../../chrome/browser/ui/browser_tabrestore.cc"
+#undef AddRestoredTab
+
+namespace {
+
+#if BUILDFLAG(ENABLE_BRAVE_WEBTORRENT)
+void MaybeLoadWebtorrent(Browser* browser,
```

Comment:
I don't think we want to duplicate the code from BraveWebTorrentNavigationThrottle, maybe you can create a static method on BraveWebTorrentNavigationThrottle or something?

Reply:
Addressed in https://github.com/brave/brave-core/pull/6085/commits/fc0d60843960b3a6fbb441219c8997c217babda4